pkg/controller/stack/stack_controller.go

```diff
@@ -313,6 +314,19 @@ func (r *ReconcileStack) Reconcile(ctx context.Context, request reconcile.Reques
 		if err != nil {
 			return reconcile.Result{}, err
 		}
+	} else if stack.ProgramRef != nil {
```

Comment:
I believe @squaremo is refactoring this selection logic in #324 already, but we should make sure to validate any mutually exclusive options rather than implicitly selecting whatever comes first in this list. i.e., the current logic will implicitly use the `GitSource` if multiple options are set.

Reply:
See https://github.com/pulumi/pulumi-kubernetes-operator/pull/324/commits/fc4f6fce787390d90af098facdad90083e7c4484#diff-6b7314d866c388df2be1614ad056c21202cdcbecace35a4496fb6eeb36ca3dc3R272 and what follows.

(I would simply wait until that's merged, and rewrite this to match when rebasing)